spec/query_associations_spec.cr

```diff
@@ -21,4 +21,96 @@ describe "Query associations" do
     posts = Post::BaseQuery.new.join_comments.comments(&.body("matching"))
     posts.results.should eq([post_with_matching_comment])
   end
+
+  it "can query associations with inner_join specified" do
+    post_with_matching_comment = PostBox.create
+    matching_comment = CommentBox
+      .new
+      .body("matching")
+      .post_id(post_with_matching_comment.id)
+      .create
+    post_without_matching_comment = PostBox.create
+    CommentBox
+      .new
+      .body("not-matching")
+      .post_id(post_with_matching_comment.id)
+      .create
+
+    posts = Post::BaseQuery.new.inner_join_comments.comments(&.body.eq("matching"))
+    posts.to_sql[0].should contain "INNER JOIN"
+    posts.results.should eq([post_with_matching_comment])
+
+    posts = Post::BaseQuery.new.inner_join_comments.comments(&.body("matching"))
+    posts.to_sql[0].should contain "INNER JOIN"
+    posts.results.should eq([post_with_matching_comment])
+  end
+
+  it "can query associations with left_join specified" do
+    post_with_matching_comment = PostBox.create
+    matching_comment = CommentBox
+      .new
+      .body("matching")
+      .post_id(post_with_matching_comment.id)
+      .create
+    post_without_matching_comment = PostBox.create
+    CommentBox
+      .new
+      .body("not-matching")
+      .post_id(post_with_matching_comment.id)
+      .create
+
+    posts = Post::BaseQuery.new.left_join_comments.comments(&.body.eq("matching"))
+    posts.to_sql[0].should contain "LEFT JOIN"
+    posts.results.should eq([post_with_matching_comment])
+
+    posts = Post::BaseQuery.new.left_join_comments.comments(&.body("matching"))
+    posts.to_sql[0].should contain "LEFT JOIN"
+    posts.results.should eq([post_with_matching_comment])
```

Comment:
I'm not sure I see the different between these two assertions. Maybe accidental duplication?

Reply:
Oh it looks like it is testing `body` and `body.eq`. I think just one should be ok since we have other spots that should catch that. It doesn't apply to joins in particular as it is just a normal criteria method